NU_PFM_M2351_NPSA_S: failing nightly in ATHandler - not supporting serial

<!--

   ************************************** WARNING **************************************

   The ciarcom bot parses this header automatically. Any deviation from the 
   template may cause the bot to automatically correct this header or may result in a 
   warning message, requesting updates.

   Please ensure all sections of the template below are filled in and no changes 
   are made to the template format. Only bugs should be raised here as issues. 
   Questions or enhancements should instead be raised on our forums:
   https://forums.mbed.com/ .

   *************************************************************************************

-->

### Description of defect

<!--
    Add detailed description of what you are reporting.
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html
-->

`NU_PFM_M2351_NPSA_S` fails with an error:

```
Compile [  9.3%]: ATHandler.cpp
[Error] ATHandler.cpp@1548,17: unknown type name 'BufferedSerial'
[Error] ATHandler.cpp@1548,49: no member named 'set_baud' in 'mbed::FileHandle'
[ERROR] ./features/cellular/framework/device/ATHandler.cpp:1548:17: error: unknown type name 'BufferedSerial'
    static_cast<BufferedSerial *>(_fileHandle)->set_baud(baud_rate);
                ^
./features/cellular/framework/device/ATHandler.cpp:1548:49: error: no member named 'set_baud' in 'mbed::FileHandle'
    static_cast<BufferedSerial *>(_fileHandle)->set_baud(baud_rate);
```

The target has this in targets.json file:
```
        "device_has_remove": [
            "SERIAL",
            "SERIAL_ASYNCH",
            "SERIAL_FC",
            "STDIO_MESSAGES",
            "LPTICKER"
        ],
```

It does not support SERIAL, therefore the object BufferedSerial does not exists. A question is how ATHandler should handle this?

@ARMmbed/mbed-os-wan  Please review

cc @ARMmbed/team-nuvoton

#### Target(s) affected by this defect ?

NU_PFM_M2351_NPSA_S

#### Toolchain(s) (name and version) displaying this defect ?

all

#### What version of Mbed-os are you using (tag or sha) ?
<!--
    For a released version please provide the release tag (this can be found as per the instructions below)

    mbed-os version can be found in /platform/mbed_version.h. The tag can be reconstructed as follows:
    mbed-os-MBED_MAJOR_VERSION.MBED_MINOR_VERSION.MBED_PATCH_VERSION
 
    Master branch is indicated by 'mbed-os-99.99.99
    
    For an issue found on Master please provide the sha being used.
-->

33563b3cfb

#### What version(s) of tools are you using. List all that apply (E.g. mbed-cli)

not relevant

#### How is this defect reproduced ? 

Build the target (test or an example) and should fail

--- a/features/cellular/framework/AT/AT_CellularDevice.cpp
+++ b/features/cellular/framework/AT/AT_CellularDevice.cpp
@@ -598,7 +598,10 @@ nsapi_error_t AT_CellularDevice::clear()
 
 nsapi_error_t AT_CellularDevice::set_baud_rate(int baud_rate)
 {
-    nsapi_error_t error = set_baud_rate_impl(baud_rate);
+    nsapi_error_t error;
+
+#if (DEVICE_SERIAL && DEVICE_INTERRUPTIN)
+    error = set_baud_rate_impl(baud_rate);
 
     if (error) {
         tr_warning("Baudrate was not changed to desired value: %d", baud_rate);
@@ -609,6 +612,12 @@ nsapi_error_t AT_CellularDevice::set_baud_rate(int baud_rate)
 
     // Give some time before starting using the UART with the new baud rate
     rtos::ThisThread::sleep_for(3000);
+#else
+    // Currently ATHandler only supports BufferedSerial based communication and
+    // if serial is disabled, baud rate cannot be set
+    tr_warn("set_baud_rate: Serial not supported");
+    error = NSAPI_ERROR_UNSUPPORTED;
+#endif
 
     return error;
 }

--- a/features/cellular/framework/device/ATHandler.cpp
+++ b/features/cellular/framework/device/ATHandler.cpp
@@ -1545,6 +1545,9 @@ void ATHandler::write_hex_string(const char *str, size_t size)
 
 void ATHandler::set_baud(int baud_rate)
 {
+#if (DEVICE_SERIAL && DEVICE_INTERRUPTIN)
     static_cast<BufferedSerial *>(_fileHandle)->set_baud(baud_rate);
+#else
+    tr_error("Device does not support BufferedSerial");
+#endif
 }
-

